mmv1/third_party/terraform/.teamcity/components/builds/build_steps.kt

```diff
@@ -77,9 +77,46 @@ fun BuildSteps.downloadTerraformBinary() {
 
 // RunSweepers runs sweepers, and relies on set build configuration parameters
 fun BuildSteps.runSweepers(sweeperStepName: String) {
-    step(ScriptBuildStep{
+    step(ScriptBuildStep {
+        name = "Compile Sweeper Test Binary"
+        workingDir = "%PACKAGE_PATH%"
+        scriptContent = """
+            #!/bin/bash
+            export TEST_FILE_COUNT=$(ls ./*_test.go | wc -l)
+            if test ${'$'}TEST_FILE_COUNT -gt "0"; then
+                echo "Compiling sweeper test binary"
+                go test -c -o test-binary
+            else
+                echo "Skipping compilation of test binary; no Go test files found"
+            fi
+        """.trimIndent()
+    })
+
+    step(ScriptBuildStep {
         name = sweeperStepName
-        scriptContent = "go test -v \"%PACKAGE_PATH%\" -run=\"%TEST_PREFIX%\" -sweep=\"%SWEEPER_REGIONS%\" -sweep-allow-failures -sweep-run=\"%SWEEP_RUN%\" -timeout 30m -json"
+        workingDir = "%PACKAGE_PATH%"
+        scriptContent = """
+            #!/bin/bash
+            if ! test -f "./test-binary"; then
+              echo "Skipping sweeper execution; file ./test-binary does not exist."
+              exit 0
```

Comment:
```suggestion
              exit 1
```

Reply:
Well i think that's a bit different since it's generic. If a test package doesn't exist that sounds like a hard failure to me. 

I'm not too worried about this though. I'm fine approving, just letting you know.